server/src/main/java/org/opensearch/common/xcontent/JsonToStringXContentParser.java

```diff
@@ -8,6 +8,7 @@
 
 package org.opensearch.common.xcontent;
 
+import org.apache.logging.log4j.util.Strings;
```

Comment:
This is probably not the right import, `org.opensearch.common`?

Reply:
Oops! You're right. I'll fix that.

We use ForbiddenAPIs, right? If so, I can add that to catch it in precommit.

Reply:
I tried forbidding all uses of `org.apache.logging.log4j.util.**` and that opened a can of worms:

```
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.env.NodeEnvironment (NodeEnvironment.java:1303)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.IndexModule$DataLocalityType (IndexModule.java:583)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.gateway.remote.RemoteClusterStateCleanupManager (RemoteClusterStateCleanupManager.java:128)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.gateway.remote.RemoteClusterStateCleanupManager (RemoteClusterStateCleanupManager.java:129)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.DeprecatedMessage (DeprecatedMessage.java, class declaration)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.tasks.consumer.SearchShardTaskDetailsLogMessage (SearchShardTaskDetailsLogMessage.java, class declaration)
Forbidden class/interface use: org.apache.logging.log4j.util.StringBuilders [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.OpenSearchMessageFieldConverter (OpenSearchMessageFieldConverter.java:76)
Forbidden class/interface use: org.apache.logging.log4j.util.StringBuilders [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.OpenSearchMessageFieldConverter (OpenSearchMessageFieldConverter.java:80)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.action.search.SearchRequestSlowLog$SearchRequestSlowLogMessage (SearchRequestSlowLog.java, class declaration)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.IndexingSlowLog$IndexingSlowLogMessage (IndexingSlowLog.java, class declaration)
Forbidden class/interface use: org.apache.logging.log4j.util.StringBuilders [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.IndexingSlowLog$IndexingSlowLogMessage (IndexingSlowLog.java:251)
Forbidden class/interface use: org.apache.logging.log4j.util.StringBuilders [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.IndexingSlowLog$IndexingSlowLogMessage (IndexingSlowLog.java:255)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.SearchSlowLog$SearchSlowLogMessage (SearchSlowLog.java, class declaration)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:87)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:92)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:93)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:100)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:113)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:132)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:138)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:149)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:168)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:180)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:183)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:195)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:210)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:221)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:230)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:257)
Forbidden class/interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.store.CompositeDirectory (CompositeDirectory.java:279)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.shard.ShardPath (ShardPath.java:208)
Forbidden interface use: org.apache.logging.log4j.util.StringBuilderFormattable [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.OpenSearchLogMessage (OpenSearchLogMessage.java, class declaration)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.xcontent.JsonToStringXContentParser (JsonToStringXContentParser.java:113)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.JsonThrowablePatternConverter (JsonThrowablePatternConverter.java:87)
Forbidden interface use: org.apache.logging.log4j.util.Supplier [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.RateLimitingFilter (RateLimitingFilter.java, method declaration of 'filter(org.apache.logging.log4j.core.Logger,org.apache.logging.log4j.Level,org.apache.logging.log4j.Marker,org.apache.logging.log4j.message.Message,java.lang.Throwable)')
Scanned 7503 class file(s) for forbidden API invocations (in 13.93s), 35 error(s).
```

Some of those do indeed look incorrect, like most of the uses of `Strings`. Maybe we just exclude that one, once we fix the others. I'll open a follow-up issue for that.

Reply:
Scoping down to just cover `org.apache.logging.log4j.util.Strings`, we get a much more manageable list of files to cover:

```
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.env.NodeEnvironment (NodeEnvironment.java:1303)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.IndexModule$DataLocalityType (IndexModule.java:583)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.gateway.remote.RemoteClusterStateCleanupManager (RemoteClusterStateCleanupManager.java:128)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.gateway.remote.RemoteClusterStateCleanupManager (RemoteClusterStateCleanupManager.java:129)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.index.shard.ShardPath (ShardPath.java:208)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.xcontent.JsonToStringXContentParser (JsonToStringXContentParser.java:113)
Forbidden class/interface use: org.apache.logging.log4j.util.Strings [use (or add) helpers to org.opensearch.common.util instead]
  in org.opensearch.common.logging.JsonThrowablePatternConverter (JsonThrowablePatternConverter.java:87)
Scanned 7503 class file(s) for forbidden API invocations (in 13.70s), 7 error(s).
```

Reply:
https://github.com/opensearch-project/OpenSearch/issues/15211